### PR TITLE
PR: Display extension type icons and create entry as files are opened in "Open recent" menu

### LINKED ISSUE #9150: Improve the "Open recent" menu
@juanis2112, I refer to this menu:

Selección_024

I would like to see the following improvements:

- [x] Create entries on it after a file is opened in the Editor. Right now, we only create an entry after the file is closed, which is odd and counter-intuitive.
- [x] Display the icon type corresponding to the file on its left. Right now we are displaying the same icon for all files, which is not so good. If a file doesn't have an icon, then we should show the same generic icon that we're displaying now.

### PR DESCRIPTION
**Pull request by juanis2112**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)
* [x] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

Display extension type icons in Recent Files menu and create entries as soon as file is opened in editor.

IMG_1349

### Issue(s) Resolved

Fixes #9150 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

Juanis2112

### CODE REVIEW

File: spyder/plugins/editor/plugin.py [hunk 1]
```diff
@@ -1653,13 +1653,13 @@ def update_recent_file_menu(self):
         """Update recent file menu"""
         recent_files = []
         for fname in self.recent_files:
-            if self.is_file_opened(fname) is None and osp.isfile(fname):
+            if osp.isfile(fname):
```
Comment (goanpeca) [inline spyder/plugins/editor/plugin.py]: @juanis2112 question, why did we remove `self.is_file_opened(fname)`?

Comment (juanis2112) [reply spyder/plugins/editor/plugin.py]: @goanpeca I understand this line is the one that doesn't let the entry to be created while the file is open. Please correct me if I'm wrong.

Comment (goanpeca) [reply spyder/plugins/editor/plugin.py]: Yes you are right, missed that part, I thought this was only about the extension!, y bad :-)

File: spyder/plugins/editor/plugin.py [hunk 2]
```diff
@@ -1653,13 +1653,13 @@ def update_recent_file_menu(self):
...
-            if self.is_file_opened(fname) is None and osp.isfile(fname):
+            if osp.isfile(fname):
                 recent_files.append(fname)
         self.recent_file_menu.clear()
         if recent_files:
             for fname in recent_files:
                 action = create_action(self, fname,
-                                       icon=ima.icon('FileIcon'),
+                                       icon=ima.get_icon_by_extension(fname, scale_factor=1.0),
                                        triggered=self.load)
```
Comment (ccordoba12) [inline spyder/plugins/editor/plugin.py]: To fix the Pep8speaks warning, please change this code to be

```python
action = create_action(
    self, fname,
    icon=ima.get_icon_by_extension(fname, scale_factor=1.0),
    triggered=self.load)
```

Comment (juanis2112) [reply spyder/plugins/editor/plugin.py]: @ccordoba12 done!

**Pull request merged**